server/gitrest/src/utils/definitions.ts

```diff
@@ -0,0 +1,60 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as git from "@fluidframework/gitresources";
+
+export interface IExternalWriterConfig {
+    enabled: boolean;
+}
+
+export interface IRepositoryManager {
+    getCommit(sha: string): Promise<git.ICommit>;
+    getCommits(sha: string, count: number, externalWriterConfig?: IExternalWriterConfig): Promise<git.ICommitDetails[]>;
+    getTree(root: string, recursive: boolean): Promise<git.ITree>;
+    getBlob(sha: string): Promise<git.IBlob>;
+    getContent(commit: string, path: string): Promise<git.IBlob>;
+    createBlob(createBlobParams: git.ICreateBlobParams): Promise<git.ICreateBlobResponse>;
+    createTree(params: git.ICreateTreeParams): Promise<git.ITree>;
+    createCommit(commit: git.ICreateCommitParams): Promise<git.ICommit>;
+    getRefs(): Promise<git.IRef[]>;
+    getRef(ref: string, externalWriterConfig?: IExternalWriterConfig): Promise<git.IRef>;
+    createRef(createRefParams: git.ICreateRefParams, externalWriterConfig?: IExternalWriterConfig): Promise<git.IRef>;
+    // eslint-disable-next-line max-len
+    patchRef(refId: string, patchRefParams: git.IPatchRefParams, externalWriterConfig?: IExternalWriterConfig): Promise<git.IRef>;
+    deleteRef(refId: string): Promise<void>;
+    getTag(tagId: string): Promise<git.ITag>;
+    createTag(tagParams: git.ICreateTagParams): Promise<git.ITag>;
+}
+
+export interface IRepositoryManagerFactory {
+    /**
+     * Create a new repository and return its manager instance.
+     */
+    create(repoOwner: string, repoName: string): Promise<IRepositoryManager>;
+    /**
+     * Open an existing repository and return its manager instance.
+     */
+    open(repoOwner: string, repoName: string): Promise<IRepositoryManager>;
+}
+
+// 100644 for file (blob)
```

Comment:
Curious, what do these comments refer to?

Reply:
TBH I'm not sure but I didn't want to remove them

Reply:
Ah, I see, so this was already there before the refactoring? I must have missed that.

Reply:
Yes, it was under gitrest/src/utils.ts (the diff is minimized by default in GitHub)